### DISCUSSION TITLE
Low FPS when using large textures

### DISCUSSION THREAD

**Discussion by denvolynets** [pixijs/pixijs#8596] [Q&A] [answered]
I need to display a large number (for example 400) of high resolution textures (3000x2600) at the same time, for this I used the ParticleContainer from the example - https://pixijs.io/examples/#/demos-basic/particle-container.js.

But in the end, I get FPS = 10-15, the more sprites, the lower the FPS, even if you turn off the animation - there is no difference, the image in idle mode also heavily loads the browser, which is why the page scroll is very freezing.
my sandbox https://codesandbox.io/s/admiring-lamarr-p8bvyf?file=/index.html

**Comment by bigtimebuddy** [comment, marked as answer]
There are typically a few things that happen to slow down FPS: 1). CPU (e.g., number of sprites), 2). GPU Uploading (e.g., number of unique textures uploaded per frame), and 3). fillrate (e.g., the amount of pixels to be painted).

In your case, you have 1 texture and 400 sprites. These are both relatively small amounts, however, your textures are massive. You're going to bump into fill rate issues for sure, which will severely limit your performance.

Let's do some quick math with your scene:

```
400 * 3000 * 2600 = 3120000000 pixels (3.12 Billion pixels)
```

This is the equivalent to ~12.2 million 16 x 16 pixel sprites. This is massive and of course you're having issues.

Without doing something overly complicated with filters/shaders, the only real option you have is to reduce the size of your textures and/or reduce the number of sprites.
